p4-16/spec/P4-16-spec.mdk

```diff
@@ -2208,16 +2208,21 @@ structField
 ~ End P4Grammar
 
 where each `typeRef` is restricted to a bit-string type (fixed or
-variable) or an integer type. This declaration introduces a new
-identifier in the current scope; the type can be referred to using
-this identifier. A header is similar to a `struct` in C,
+variable), a fixed-width signed integer type, a boolean type, or a struct that itself contains other 
+struct fields, nested arbitrarily, as long as all of the "leaf" types are
+are `bit<W>`, `int<W>`, a Serializable Enum, or a `bool`.  If a `bool` is used
+inside a P4 header, all implementations encode the `bool` as a one bit long field. 
+This declaration introduces a new identifier in the current scope; the 
```

Comment:
Please be more precise on what is `This declaration`, what is `current scope`, and what is `this identifier`.

Reply:
Please see the original p4-16 spec at the section at this URL

https://p4.org/p4-spec/docs/P4-16-v1.1.0-spec.html#sec-header-types

The declaration is more clear there.  In this PR, there is more text added and thus the "declaration" , current scope and this identifier is obscured.  I will have to think about this one,